### PR TITLE
Gender selector

### PR DESCRIPTION
**Pull request by edlu77**

### CODE REVIEW

**Comment by bherr2** [comment]
Failed to pass linting

File: projects/ccf-rui/src/app/shared/components/gender-selector/gender-selector.component.theme.scss [hunk 1]
```diff
@@ -0,0 +1,29 @@
+@use '~@angular/material/theming' as *;
+
+@mixin theme($theme) {
+  $foreground: map-get($theme, foreground);
+  $background: map-get($theme, background);
+  $primary: map-get($theme, primary);
+
+  .gender-slider {
```
Comment (axdanbol) [inline projects/ccf-rui/src/app/shared/components/gender-selector/gender-selector.component.theme.scss:8]: These rules become global non-encapsulated therefore they need to be wrapped inside a css selector for this specific component. Check right-sidebar code + theme on how it is done.

File: projects/ccf-rui/src/app/shared/components/organ-selector/organ-selector.component.theme.scss [hunk 2]
```diff
@@ -8,7 +8,7 @@
   .carousel {
     color: mat-color($foreground, text);
     border: 1px solid mat-color($foreground, borders);
-    background-color: mat-color($background, dialog);
+    background-color: white;
```
Comment (axdanbol) [inline projects/ccf-rui/src/app/shared/components/organ-selector/organ-selector.component.theme.scss:11]: This should refer to a variable in the background (you can add additional variables in the theme files if needed) not be hardcoded!

File: projects/ccf-rui/src/app/shared/components/side-selector/side-selector.component.theme.scss [hunk 3]
```diff
@@ -0,0 +1,29 @@
+@use '~@angular/material/theming' as *;
+
+@mixin theme($theme) {
+  $foreground: map-get($theme, foreground);
+  $background: map-get($theme, background);
+  $primary: map-get($theme, primary);
+
+  .side-slider {
+    .mat-slide-toggle-bar {
+      background-color: rgba(25, 118, 210, 0.54) !important;
```
Comment (axdanbol) [inline projects/ccf-rui/src/app/shared/components/side-selector/side-selector.component.theme.scss:10]: These colors should come from the theme!! Not hardcoded! Not to mention the use of !important

File: projects/ccf-rui/src/themes/common/_theming.scss [hunk 4]
```diff
@@ -27,6 +29,8 @@
   @include rightSidebar.theme($theme);
   @include blockSizeInput.theme($theme);
   @include slicesInput.theme($theme);
+  @include sideSelector.theme($theme);
```
Comment (axdanbol) [inline projects/ccf-rui/src/themes/common/_theming.scss:32]: Minor thing. Please move down to after organ-selector. I am trying to keep this file a bit tidier by grouping things together based on ui location.

File: projects/ccf-rui/src/app/shared/components/gender-selector/gender-selector.component.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
+import { Component, Input, Output, EventEmitter } from '@angular/core';
```
Comment (axdanbol) [inline projects/ccf-rui/src/app/shared/components/gender-selector/gender-selector.component.ts:1]: This entire component and side-selector component are duplicates. Please create a single generic component (named `labeled-side-toggle` perhaps)

File: projects/ccf-rui/src/app/shared/components/gender-selector/gender-selector.component.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+   */
+  @Output() genderChanged = new EventEmitter<string>();
+
+  /**
+   * Updates and emits the currently selected gender
+   * @param selection The current toggle state
+   */
+  updateGender(selection: boolean): void {
+    this.isMale = !selection;
+    this.genderChanged.emit(this.isMale ? 'male' : 'female');
```
Comment (axdanbol) [inline projects/ccf-rui/src/app/shared/components/gender-selector/gender-selector.component.ts:34]: This branch does not have full test coverage! You need to test both when it is set and not set.

**Pull request merged**